python_modules/libraries/dagster-duckdb-pandas/dagster_duckdb_pandas/duckdb_pandas_type_handler.py

```diff
@@ -62,7 +63,16 @@ def load_input(
         self, context: InputContext, table_slice: TableSlice, connection
     ) -> pd.DataFrame:
         """Loads the input as a Pandas DataFrame."""
-        return connection.execute(DuckDbClient.get_select_statement(table_slice)).fetchdf()
+        try:
```

Comment:
alternative logic here would be 
```
if table_slice.is_self_dependent and is_first_partition:
    return pd.DataFrame()
return connection.execute(...)
```

but i'm not sure how to figure out the `is_first_partition` part yet. I could compare the partition key with the start of the partition definition, but static partitions are a bit more complicated